test/correctness/simd_op_check.cpp

```diff
@@ -1493,7 +1495,12 @@ struct Test {
 
         check("vadd(v*.b,v*.b)", hvx_width/1, u8_1 + u8_2);
         check("vadd(v*.h,v*.h)", hvx_width/2, u16_1 + u16_2);
-        check("vadd(v*.w,v*.w)", hvx_width/4, u32_1 + u32_2);
+        if (is_v62) {
+            // v62 - 32 bit saturating add
+            check("vadd(v*.uw,v*.uw):sat", hvx_width/4, u32_1 + u32_2);
```

Comment:
I dont think this is the right pattern to match. u32_1 + u32_2 is adding two unsigned 32 bit values. Unsigned overflow is defined (modulo). saturated addition clamps the add to the UINT_MAX instead. You should really be matching this pattern u32_sat(u64(u32_1) + u64(u32_1));